src/python/ensembl/database/dbconnection.py

```diff
@@ -0,0 +1,202 @@
+# .. See the NOTICE file distributed with this work for additional information
+#    regarding copyright ownership.
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#        http://www.apache.org/licenses/LICENSE-2.0
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+"""Database connection handler.
+
+This module provides the main class to connect to and access databases. The connection will provide ORM-less,
+that is, the data can only be accessed via SQL queries (see example below).
+
+Typical usage example::
+
+    dbc = DBConnection('mysql://ensro@mysql-server:4242/mydb')
+    # You can access the database data via sql queries, for instance:
+    results = dbc.execute('SELECT * FROM my_table;')
+    # Or via a connection in a transaction manner:
+    with dbc.begin() as conn:
+        results = conn.execute('SELECT * FROM my_table;')
+
+"""
+
+import contextlib
+from typing import Dict, List, TypeVar
+
+import sqlalchemy
+from sqlalchemy import create_engine, event, text
+from sqlalchemy.orm.session import Session
+from sqlalchemy.sql import select
+
+
+# Create the Query type as an alias for all the possible types an SQL query can be stored into
+Query = TypeVar('Query', str, sqlalchemy.sql.expression.ClauseElement)
+# Create the URL type as an alias for all the possible types an URL can be stored into
+URL = TypeVar('URL', str, sqlalchemy.engine.url.URL)
+
+
+class DBConnection:
+    """Database connection handler, providing also the database's schema and properties.
+
+    Args:
+        url: URL to the database, e.g. ``mysql://user:passwd@host:port/my_db``.
+
+    """
+    def __init__(self, url: URL) -> None:
+        self._engine = create_engine(url)
+        self.load_metadata()
+
+    def __repr__(self) -> str:
+        """Returns a string representation of this object."""
+        return f'{self.__class__.__name__}({self.url!r})'
+
+    def load_metadata(self) -> None:
```

Comment:
Is this method intended to be called by us (the Python developers) ? If no, what about hiding it with `__` or embedding it into `__init__` ?

Reply:
Yes, it needs to be called in `unittest` **after** the database has been loaded, so the metadata information is updated accordingly, so "the rules" state it shouldn't be hidden.